prov/rxm/src/rxm_cq.c

```diff
@@ -480,10 +487,15 @@ static ssize_t rxm_rndv_xfer(struct rxm_ep *rxm_ep, struct fid_ep *msg_ep,
 		if (ret)
 			return ret;
 		total_len -= copy_len;
-		ret = rxm_ep->rndv_ops->xfer(msg_ep, iov, desc, count, 0,
-			       remote_hdr->iov[i].addr, remote_hdr->iov[i].key,
-			       context);
 
+		if (conn_sts)
+			ret = conn_sts;
+		else
```

Comment:
I'm lost on the purpose of this check.  Why did we continue to here?  I think conn_sts (not sure what sts means btw) is -FI_EAGAIN.  Splitting the check of conn_sts so far from where it was set is confusing.

Reply:
@shefty - It was just to be more efficient and not repeatedly call the function that checks if connectivity has been established within a loop. I will look at reworking this code to see if I can make it easier to follow (if I decide to move the check for duplex connectivity further up in the processing I will remove it). Thanks.